sky/resources.py

```diff
@@ -1162,6 +1163,13 @@ def _try_validate_image_id(self) -> None:
         Raises:
             ValueError: if the attribute is invalid.
         """
+
+        if self._image_id is None and isinstance(self._cloud, clouds.GCP):
+            if self._network_tier == resources_utils.NetworkTier.BEST:
+                self._image_id = {
+                    self._region: gcp_constants.GCP_GPU_DIRECT_IMAGE_ID
+                }
```

Comment:
If the user specifies some custom `image_id` that is not docker image, we should probably error out, and tell the user that the image id has to be a docker image, and recommended to leave it empty?